libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1188,35 +1173,31 @@ Json::Value StandardCompiler::compileSolidity(StandardCompiler::InputsAndSetting
 		errors.append(formatErrorWithException(
 			compilerStack,
 			_exception,
-			Error::Severity::Error,
-			"SMTLogicException",
+			Error::Type::SMTLogicException,
 			"general",
 			"SMT logic exception"
 		));
 	}
 	catch (util::Exception const& _exception)
 	{
 		errors.append(formatError(
-			Error::Severity::Error,
-			"Exception",
+			Error::Type::Exception,
 			"general",
 			"Exception during compilation: " + boost::diagnostic_information(_exception)
 		));
 	}
 	catch (std::exception const& _exception)
 	{
 		errors.append(formatError(
-			Error::Severity::Error,
-			"Exception",
+			Error::Type::Exception,
 			"general",
 			"Unknown exception during compilation: " + boost::diagnostic_information(_exception)
 		));
 	}
 	catch (...)
 	{
 		errors.append(formatError(
-			Error::Severity::Error,
-			"Exception",
+			Error::Type::InternalCompilerError,
```

Comment:
It was `"Exception"` but you changed this to `InternalCompilerError`. Was that intentional?

It's not really wrong - it's debatable which one is more fitting here - but it changes current behavior. I think we should consider removing `Type::Exception` and just using the ICE; maybe also squash some other redundant types. I don't want to get into that now though so I'd leave this for a separate PR.

Reply:
Restored it to `Error::Exception` . 

We can look at removing these redundant error types in another PR. 
@cameel Should I make another issue out of it ?

Reply:
I'd wait with a PR until we're done with this one.

Would be best if you could first propose which ones you would eliminate/merge. `Exception` is an obvious one but maybe we can get rid of some more. Then we can discuss on the channel whether to go on with this (and how breaking it is).

Reply:
> Restored it to Error::Exception .

Not sure what was decided here, but it was not changed.
Although, it does not seem to affect any test output....

Reply:
Changed now. 

Will make another PR to remove redundant error types, once this is ready for merging.

Reply:
> The change is resulting in an error on the CI.

Do you mean [this one in `t_ubu_soltest_all`](https://app.circleci.com/pipelines/github/ethereum/solidity/25863/workflows/23ed2022-7e2e-481f-b79a-541c5f592463/jobs/1141890/parallel-runs/4)? It looks completely unrelated. It's in one of the tests that create files/dirs on disk and it could be that running stuff in parallel is messing with it.

Is it reproducible?

Reply:
Yes. It was indeed completely unrelated. 
A mere rerun ended up passing the test